python/src/uagents/types.py

```diff
@@ -55,6 +56,30 @@ class RestHandlerDetails(BaseModel):
     response_model: Type[Union[Model, BaseModel]]
 
 
+class AgentGeolocation(BaseModel):
+    latitude: Annotated[
+        float,
+        Field(strict=True, ge=-90, le=90, allow_inf_nan=False),
+    ]
+    longitude: Annotated[
+        float,
+        Field(strict=True, ge=-180, le=180, allow_inf_nan=False),
+    ]
+    radius: Annotated[
+        float,
+        Field(strict=True, ge=0, allow_inf_nan=False),
+    ] = 0
+
+    @field_serializer("latitude", "longitude")
+    def serialize_precision(self, val: float) -> float:
+        """Round the latitude and longitude to 6 decimal places."""
+        return float(f"{val:.6f}")
```

Comment:
What's the reason for this?

Reply:
A precision of 6 digits after the comma equates to 10cm so I think that should be enough.
There is a funny [comparison](https://xkcd.com/2170/) from xkcd for that 😄